index.tsx

```diff
@@ -0,0 +1,346 @@
+import * as React from 'react';
+import {
+  View,
+  StyleSheet,
+  TouchableOpacity,
+  BackHandler,
+  NativeModules,
+  Platform,
+  PermissionsAndroid,
+  ViewStyle,
+  StyleProp,
+} from 'react-native';
+
+import Overlay from './components/Overlay';
+import Sheet from './components/Sheet';
+import Button from './components/Button';
+
+const styles = StyleSheet.create({
+  actionSheetContainer: {
+    flex: 1,
+    paddingTop: 10,
+    paddingBottom: 0,
+    justifyContent: 'flex-end',
+    backgroundColor: 'rgba(0, 0, 0, 0.5)',
+  },
+  buttonContainer: {
+    overflow: 'hidden',
+    backgroundColor: 'white',
+    paddingBottom: 5,
+    paddingTop: 5,
+  },
+});
+
+type Props = {
+  visible: boolean;
+  onCancel: () => void;
+  style?: StyleProp<ViewStyle>;
+  overlayStyle?: StyleProp<ViewStyle>;
+};
+
+const shareSheetStyle = { flex: 1 };
+
+class ShareSheet extends React.Component<Props> {
+  componentDidMount() {
+    this.backButtonHandler = this.backButtonHandler.bind(this);
+    // @ts-ignore is this on purpose or maybe wrong?
+    BackHandler.addEventListener('backPress', this.backButtonHandler);
+  }
+
+  componentWillUnmount() {
+    // @ts-ignore is this on purpose or maybe wrong?
+    BackHandler.removeEventListener('backPress', this.backButtonHandler);
+  }
+
+  backButtonHandler() {
+    if (this.props.visible) {
+      this.props.onCancel();
+      return true;
+    }
+    return false;
+  }
+  render() {
+    const { style = {}, overlayStyle = {}, visible, ...props } = this.props;
+    return (
+      <Overlay visible={visible} {...props}>
```

Comment:
spread was overwriting prop always

Reply:
Do you think It would be better to avoid the `props` to override the `visible`?

```suggestion
      <Overlay  {...props} visible={visible}>
```

Reply:
ts reported it as an error, I'd rather not touch any logic tho, so yeah, will try to go back to overwriting it